scripts/google/covid_mobility/covidmobility.py

```diff
@@ -0,0 +1,235 @@
+    """Main method for the covid_mobility script.
+
+    Args:
+        path_to_store_data (str): Defaults to './input/input_data.csv'.
+        export_to (str): Defaults to './output/covid_mobility_output.mcf'.
+    """
+
+    # URL to download the data from Google Mobility site.
+    url: str = "https://www.gstatic.com/covid19/mobility/" + \
+               "Global_Mobility_Report.csv"
+
+    # Download CSV data from url.
+    download_data(url=url, download_as=input_path)
+
+    # Convert the CSV data to MCF.
+    csv_to_mcf(input_path, output_path)
+
+
+def csv_to_mcf(input_path: str, output_path: str) -> None:
+    """Converts the Mobility data to MCF.
+
+    Args:
+        input_path (str): The path to the CSV file containing the data.
+        output_path (str): The path to write the output MCF file.
+    """
+
+    visited_dcids: set = set()
+
+    # If output file already exists, remove it.
+    if ospath.exists(output_path):
+        remove(output_path)
+
+    f_input = open(input_path, 'r')
+    f_output = open(output_path, 'a+')
+
+    csv_reader: DictReader = DictReader(f_input)
+
+    for row in csv_reader:
+        # Get the region names.
+        # If the column doesn't exist, skip the row.
+        try:
+            sub_region1: str = row['sub_region_1']
+            sub_region2: str = row['sub_region_2']
+            country_code: str = row['country_region_code']
+            date = row['date']
+        except KeyError:
+            continue
+
+        # Convert the region name to a dcid/geoId.
+        region_dcid: str = get_region_dcid(sub_region2,
+                                           sub_region1,
+                                           country_code)
+
+        # If no dcid, skip the row.
+        if not region_dcid:
+            continue
+
+        # If no date, skip the row.
+        if not date:
+            continue
+
+        # Iterate through all places in the row.
+        for place in PLACE_CATEGORIES:
+            # Type of place using Schema notation.
+            schema_place = PLACE_CATEGORIES[place]
+
+            population_id = f"{region_dcid}_{schema_place}"
+            population_id = convert_to_ascii(population_id)
+
+            observation_id = f"{population_id}_{date}"
+
+            # If this is the first time vieweing this dcid.
+            # Write the population node for the place.
+            if region_dcid not in visited_dcids:
+                f_output.write(
+                    f"Node: {population_id}\n"
+                    "typeOf: schema:StatisticalPopulation\n"
+                    f"location: dcid:{region_dcid}\n"
+                    "populationType: dcs:PlaceVisitEvent\n"
+                    f"placeCategory: dcs:{schema_place}\n\n")
+
+            # Get the value for the current place.
+            value = row[place]
+
+            # If the value is None, skip the row.
+            if not value:
+                continue
```

Comment:
nit: You could do this check before writing on line 113 so that its cleaner that we write a StatPop only when there's a value.

Reply:
Done